pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2098,7 +2098,7 @@ func (wf *Workflow) GetStoredTemplate(scope ResourceScope, resourceName string,
 		return nil
 	}
 	if tmpl, ok := wf.Status.StoredTemplates[tmplID]; ok {
-		return &tmpl
+		return tmpl.DeepCopy()
```

Comment:
When stored templates are returned here they are returned as a reference. If any piece of the code modifies them thinking they are a value, the changes will be persisted when `woc.persistUpdates()` is called. Ensure we only return values so that stored templates are immutable, as desired.